vault/vault.go

```diff
@@ -85,17 +86,22 @@ func New(
 	} else {
 		backendPath = DefaultBackendPath
 	}
-
-	isKvV2, err := isKvV2(client, backendPath)
-	if err != nil {
-		return nil, err
+	var isBackendV2 bool
+	backendVersion := getVaultParam(secretConfig, VaultBackendVersionKey)
+	if backendVersion != "2" {
+		isBackendV2, err = isKvV2(client, backendPath)
```

Comment:
This should be called only if the env variable is not set or empty.
If the kv backend version is 1, even then we should not invoke sys/mounts